kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/AdaptiveBigQueryWriter.java

```diff
@@ -74,16 +80,26 @@ public AdaptiveBigQueryWriter(BigQuery bigQuery,
   public void performWriteRequest(PartitionedTableId tableId,
                                   List<InsertAllRequest.RowToInsert> rows,
                                   String topic) {
-    InsertAllRequest request = createInsertAllRequest(tableId, rows);
-    InsertAllResponse writeResponse = bigQuery.insertAll(request);
-    // Should only perform one schema update attempt; may have to continue insert attempts due to
-    // BigQuery schema updates taking up to two minutes to take effect
-    if (writeResponse.hasErrors()
-        && onlyContainsInvalidSchemaErrors(writeResponse.getInsertErrors())) {
-      try {
-        schemaManager.updateSchema(tableId.getBaseTableId(), topic);
-      } catch (BigQueryException exception) {
-        throw new BigQueryConnectException("Failed to update table schema for: " + tableId.getBaseTableId(), exception);
+    InsertAllResponse writeResponse = null;
+    InsertAllRequest request = null;
+
+    try {
+      request = createInsertAllRequest(tableId, rows);
+      writeResponse = bigQuery.insertAll(request);
+      // Should only perform one schema update attempt; may have to continue insert attempts due to
+      // BigQuery schema updates taking up to two minutes to take effect
+      if (writeResponse.hasErrors()
+              && onlyContainsInvalidSchemaErrors(writeResponse.getInsertErrors())) {
+        attemptSchemaUpdate(tableId, topic);
+      }
+    } catch (BigQueryException e) {
+      if (isTableMissingSchema(e)) {
+        attemptSchemaUpdate(tableId, topic);
+
+        // If the table was missing its schema, we never received a writeResponse
+        writeResponse = bigQuery.insertAll(request);
```

Comment:
Actually, I wonder if this should be reformatted a bit.

If we have a write response with invalid schema errors, then we retry the insert with an attempt count (lines 106-121) because the schema update might be delayed.  I assume this is also a possibility if there is no schema all all.

So, I feel like this insertAll should be covered in that block, and not here.

Reply:
If we have no schema, the `writeResponse` will be null so we'll run into issues during the while loop's `writeResponse.hasErrors` check. Just to clarify, are you thinking we should update the while loop to handle the case where the writeResponse is null? That seems reasonable to me but want to verify before implementing

Reply:
yeah, I think the retry-after-schema-related-failure while loop should be updated to cover the case of this type of schema-related failure.